tracer/src/Datadog.Trace/Ci/CiEnvironment/GitInfoProvider.cs

```diff
@@ -0,0 +1,76 @@
+// <copyright file="GitInfoProvider.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Datadog.Trace.Ci.CiEnvironment;
+
+internal abstract class GitInfoProvider : IGitInfoProvider
+{
+    protected static DirectoryInfo? GetParentGitFolder(string? innerFolder)
+    {
+        if (string.IsNullOrEmpty(innerFolder))
+        {
+            return null;
+        }
+
+        var dirInfo = new DirectoryInfo(innerFolder);
+        while (dirInfo != null)
+        {
+            var gitDirectories = dirInfo.GetDirectories(".git");
```

Comment:
Is there a chance this could throw if we move to a folder we don't have permission to enumerate? Do we care - i.e. if this is caught higher up that might be fine 🤷‍♂️

Reply:
good one, I changed it in [d107ce2](https://github.com/DataDog/dd-trace-dotnet/pull/6259/commits/d107ce21648aec65ae5ba0a696ec0c908d74e0c6)